src/ts/content/entities/bookForm/data/uniqueData/uniqueFormElement.ts

```diff
@@ -0,0 +1,42 @@
+import {match} from "../../../../../common/util/match";
+import {FormData} from "../../types";
+
+type Predicate = (element) => boolean;
+type FromFormData = (formData: FormData, element) => {value: string; checked: boolean};
+type FromElement = (formData: FormData, element) => void;
+
+interface UniqueFormElementDescriptor {
+	isFormData: () => readonly [Predicate, () => true];
+	fromFormData: (formData: FormData) => readonly [Predicate, (element) => {value: string; checked: boolean}];
+	fromElement: (formData: FormData) => readonly [Predicate, (element) => void];
+}
+
+interface UniqueFormElementOptions {
+	predicate: Predicate;
+	fromFormData: FromFormData;
+	fromElement: FromElement;
+}
+
+const uniqueFormElement = ({
+	predicate,
+	fromElement,
+	fromFormData,
+}: UniqueFormElementOptions): UniqueFormElementDescriptor => ({
+	isFormData: () => [predicate, () => true] as const,
+	fromFormData: (formData: FormData) => [predicate, (element) => fromFormData(formData, element)] as const,
+	fromElement: (formData: FormData) => [predicate, (element) => fromElement(formData, element)] as const,
+});
+
+const matchFactoryFromDescriptors =
+	(...descriptors: UniqueFormElementDescriptor[]) =>
+	<T>(
+		kind: Exclude<keyof UniqueFormElementDescriptor, "predicate">,
+		otherwise: (formData: FormData, element: Element) => T
+	) =>
+	(formData: FormData, element: Element) =>
+		descriptors
+			.reduce((acc, descriptor) => acc.case(...descriptor[kind](formData)), match(element))
+			.default(() => otherwise(formData, element))
+			.yield();
```

Comment:
The idea is just to remove writing out the `match` and the cases three times. And if we add a new case, it will now automatically update all the `match`es.